code/lib/preview-web/src/PreviewWeb.tsx

```diff
@@ -49,31 +50,40 @@ function focusInInput(event: Event) {
   return /input|textarea/i.test(target.tagName) || target.getAttribute('contenteditable') !== null;
 }
 
-type PossibleRender<TFramework extends AnyFramework> =
+type PossibleRender<TFramework extends Framework> =
   | StoryRender<TFramework>
   | TemplateDocsRender<TFramework>
   | StandaloneDocsRender<TFramework>;
 
-function isStoryRender<TFramework extends AnyFramework>(
+function isStoryRender<TFramework extends Framework>(
   r: PossibleRender<TFramework>
 ): r is StoryRender<TFramework> {
   return r.type === 'story';
 }
 
-export class PreviewWeb<TFramework extends AnyFramework> extends Preview<TFramework> {
-  urlStore: UrlStore;
+export class PreviewWeb<TFramework extends Framework> extends Preview<TFramework> {
+  selectionStore: SelectionStore;
 
-  view: WebView;
+  view: View<TFramework['rootElement']>;
 
   currentSelection?: Store_Selection;
 
   currentRender?: PossibleRender<TFramework>;
 
-  constructor() {
+  constructor({
+    // I'm not quite sure how to express this -- if you don't pass a view, you need to ensure
+    // this is a PreviewWeb<..., HTMLElement>. If you pass one, it can be parameterized however you want,
+    // as long as it is consistent.
+    view = new WebView() as any,
+    selectionStore = new UrlStore(),
+  }: {
+    view?: View<TFramework['rootElement']>;
+    selectionStore?: SelectionStore;
+  } = {}) {
     super();
 
-    this.view = new WebView();
-    this.urlStore = new UrlStore();
+    this.view = view;
+    this.selectionStore = selectionStore;
```

Comment:
I know this came up in our pairing session at some point, but then we found a way to get the test to work without the need for this. Do we still actually have a use-case for this?

Reply:
@ndelangen - sorry I don't understand the question. What is "this"? If you mean parameterizing the webview/selection store, then yes there is a need for this! React Native needs to override them and that's the point of this PR.